Missing DEPENDS causes strange error

| | |
|:---|:---|
|**_F´ Version_**| 4.x|
|**_Affected Component_**|   |
---
## Problem Description

If a module `CMakeLists.txt` has an empty `DEPENDS` section in the `register_fprime_library` function call, the `fprime-util generate` step will emit a bizarre error:

```
CMake Error at lib/fprime/cmake/utilities.cmake:788 (message):
   ----------------------------------------
   F Prime/CMake target 'TRUE' not available to deployment 'SomeDeployment'. 'TRUE' must:
       1. Must be defined somewhere in the F Prime project
       2. Must be defined before 'SomeDeployment' deployment (register_fprime_deployment)
   'TRUE' is undefined, or included via `add_fprime_subdirectory` after `register_fprime_deployment`. in:
       /home/tcanham/source/SomeDirectory/Something/CMakeLists.txt
   ----------------------------------------

Call Stack (most recent call first):
  lib/fprime/cmake/utilities.cmake:737 (fprime_cmake_clear_message)
  lib/fprime/cmake/target/target.cmake:75 (fprime_cmake_fatal_error)
  lib/fprime/cmake/target/target.cmake:136 (check_unknown_links)
  lib/fprime/cmake/target/target.cmake:181 (setup_single_target)
  lib/fprime/cmake/API.cmake:161 (setup_module_targets)
  lib/fprime/cmake/API.cmake:346 (fprime_attach_custom_targets)
  SomeDeploymentDirectory/CMakeLists.txt:19 (register_fprime_deployment)
```

(Fake directories and names were inserted into the error text)

## Context / Environment

Execute `fprime-util version-check` and share the output.

```
Operating System: Linux
CPU Architecture: x86_64
Platform: Linux-5.15.133.1-microsoft-standard-WSL2-x86_64-with-glibc2.39
Python version: 3.12.3
CMake version: 3.26.0
Pip version: 26.1.1
Pip packages:
    fprime-tools==4.2.1
    fprime-gds==4.2.1
    fprime-fpp==3.2.0
Project submodules:
    https://github.com/nasa/fprime.git @ v4.2.2
```

## How to Reproduce

1.  Take any module directory with no `DEPENDS` in the `register_fprime_library`
2.  Add `DEPENDS` with no string
3.  Run fprime-util generate

## Expected Behavior

It would be nice to have an error that is more informative like "Empty DEPENDS, dummy!"

--- a/cmake/module.cmake
+++ b/cmake/module.cmake
@@ -78,6 +78,7 @@ function(fprime__process_module_setup FPRIME_MODULE_TYPE ADDITIONAL_CONTROL_SETS
     set(FILE_CONTROL_SETS "HEADERS" "SOURCES" "AUTOCODER_INPUTS")
     set(FPRIME_CMAKE_ADD_OPTIONS "WIN32" "MACOSX_BUNDLE" "OBJECT" "INTERFACE" "IMPORTED" "ALIAS" "GLOBAL"
         "STATIC" "SHARED" "MODULE" "EXCLUDE_FROM_ALL")
+    set(FPRIME_FLAG_CONTROL_SETS ${FPRIME_CMAKE_ADD_OPTIONS} "BASE_CONFIG" "UT_AUTO_HELPERS" "INCLUDE_GTEST")
     # Set module name as passed in, then defaulting to FPRIME_CURRENT_MODULE
     if (${INPUT_COUNT} GREATER 0 AND NOT FIRST_ARGUMENT IN_LIST CONTROL_SETS)
         list(POP_FRONT INPUT_ARGUMENTS MODULE_NAME)
@@ -155,7 +156,7 @@ function(fprime__process_module_setup FPRIME_MODULE_TYPE ADDITIONAL_CONTROL_SETS
         # between this and other control words.
         elseif(ARGUMENT IN_LIST CONTROL_SETS)
             # Check for control words that are zero-argument (flags) and set them to true
-            if (DEFINED CURRENT_LIST_NAME AND NOT DEFINED "LIST_${CURRENT_LIST_NAME}")
+            if (DEFINED CURRENT_LIST_NAME AND CURRENT_LIST_NAME IN_LIST FPRIME_FLAG_CONTROL_SETS AND NOT DEFINED "LIST_${CURRENT_LIST_NAME}")
                 set("LIST_${CURRENT_LIST_NAME}" TRUE)
             endif()
             set(CURRENT_LIST_NAME "${ARGUMENT}")
@@ -173,7 +174,7 @@ function(fprime__process_module_setup FPRIME_MODULE_TYPE ADDITIONAL_CONTROL_SETS
         endif()
     endforeach()
     # Check for control words that are zero-argument (flags) and set them to true
-    if (DEFINED CURRENT_LIST_NAME AND NOT DEFINED "LIST_${CURRENT_LIST_NAME}")
+    if (DEFINED CURRENT_LIST_NAME AND CURRENT_LIST_NAME IN_LIST FPRIME_FLAG_CONTROL_SETS AND NOT DEFINED "LIST_${CURRENT_LIST_NAME}")
         set("LIST_${CURRENT_LIST_NAME}" TRUE)
     endif()
     # Update caller scope with the new variables
@@ -273,4 +274,3 @@ function(fprime__internal_add_build_target_helper TARGET_NAME TYPE SOURCES AUTOC
         append_list_property("${REQUIRES_IMPLEMENTATIONS}" GLOBAL PROPERTY FPRIME_REQUIRED_IMPLEMENTATIONS)
     endif()      
 endfunction()
-

--- a/cmake/test/data/test-fprime-library/TestLibrary/TestComponent/CMakeLists.txt
+++ b/cmake/test/data/test-fprime-library/TestLibrary/TestComponent/CMakeLists.txt
@@ -3,6 +3,7 @@ register_fprime_module(
     "${CMAKE_CURRENT_LIST_DIR}/TestComponent.fpp"
   SOURCES
     "${CMAKE_CURRENT_LIST_DIR}/TestComponent.cpp"
+  DEPENDS
   HEADERS
     "${CMAKE_CURRENT_LIST_DIR}/TestComponent.hpp"
 )

